openvdb/points/PointMask.h

```diff
@@ -0,0 +1,384 @@
+            }
+            else {
+                // otherwise increment existing values
+                for (auto iter = leaf->cbeginValueOn(); iter; ++iter) {
+                    newLeaf->modifyValue(iter.getCoord(), SumOp(*iter));
+                }
+            }
+        }
+    }
+
+private:
+    TreeT& mTree;
+}; // struct GridCombinerOp
+
+
+/// @brief Compute scalar grid from PointDataGrid while evaluating the point filter
+template <typename GridT, typename PointDataGridT, typename FilterT>
+struct PointsToScalarOp
+{
+    using TreeT = typename GridT::TreeType;
+    using LeafManagerT = typename tree::LeafManager<TreeT>;
+    using ValueT = typename TreeT::LeafNodeType::ValueType;
+
+    PointsToScalarOp(const PointDataGridT& grid,
+                  const FilterT& filter)
+        : mPointDataAccessor(grid.getConstAccessor())
+        , mFilter(filter) { }
+
+    void operator()(const typename LeafManagerT::LeafRange& range) const {
+
+        for (auto leaf = range.begin(); leaf; ++leaf) {
+
+            const auto* const pointLeaf =
+                mPointDataAccessor.probeConstLeaf(leaf->origin());
+
+            // assumes matching topology
+            assert(pointLeaf);
+
+            for (auto value = leaf->beginValueOn(); value; ++value) {
+                const Index64 count = points::iterCount(
+                    pointLeaf->beginIndexVoxel(value.getCoord(), mFilter));
+                if (count > Index64(0)) {
+                    value.setValue(static_cast<ValueT>(count));
+                }
+                else {
+                    // disable any empty voxels
+                    value.setValueOn(false);
+                }
+            }
+        }
+    }
+
+private:
+    const typename PointDataGridT::ConstAccessor mPointDataAccessor;
+    const FilterT& mFilter;
+}; // struct PointsToScalarOp
+
+
+/// @brief Compute scalar grid from PointDataGrid using a different transform
+///        and while evaluating the point filter
+template <typename GridT, typename PointDataGridT, typename FilterT>
+struct PointsToTransformedScalarOp
+{
+    using LeafManagerT = tree::LeafManager<const typename PointDataGridT::TreeType>;
+    using ValueType = typename GridT::TreeType::LeafNodeType::ValueType;
+    using HandleT = AttributeHandle<Vec3f>;
+    using CombinableT = typename GridCombinerOp<GridT>::CombinableT;
+
+    PointsToTransformedScalarOp(const math::Transform& newTransform,
+                         const math::Transform& transform,
+                         const FilterT& filter,
+                         CombinableT& combinable)
+        : mNewTransform(newTransform)
+        , mTransform(transform)
+        , mFilter(filter)
+        , mCombinable(combinable) {}
+
+    void operator()(const typename LeafManagerT::LeafRange& range) const
+    {
+        bool exists;
+        auto& grid = mCombinable.local(exists);
+        if (!exists) {
+            grid.setTransform(mNewTransform.copy());
+        }
```

Comment:
You don't need to do this - as long as mNewTransform is used correctly below you can simply set the transform once on the final combined result

Reply:
Good point, fixed.